File: dataset_reduzido/avatar/Python/Code/atcoder_ABC140_D.py
from _collections import deque
def parser ( ) :
    while 1 :
        data = list ( input ( ).split ( ' ' ) )
        for number in data :
            if len ( number ) > 0 :
                yield ( number )
input_parser = parser ( )
def gw ( ) :
    global input_parser
    return next ( input_parser )
def gi ( ) :
    data = gw ( )
    return int ( data )
MOD = int ( 1e9 + 7 )
import numpy
import scipy
from collections import deque
N = gi ( )
K = gi ( )
S = gw ( )
seg_cnt = 0
for i in range ( 0 , N - 1 ) :
    if S [ i ] != S [ i + 1 ] :
        seg_cnt += 1
cur_happy = N - 1 - seg_cnt
if 2 * K >= seg_cnt :
    ans = N - 1
else :
    ans = cur_happy + 2 * K
print ( ans )

File: output/StarCoder/avatar/Python/Java/atcoder_ABC140_D.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int K = scanner.nextInt();
        String S = scanner.next();
        int segCnt = 0;
        for (int i = 0; i < N - 1; i++) {
            if (!S.substring(i, i+1).equals(S.substring(i+1, i+2))) {
                segCnt++;
            }
        }
        int curHappy = N - 1 - segCnt;
        long MOD = (long) 1e9 + 7;
        long ans = 2 * K >= segCnt ? N - 1 : curHappy + 2 * K;
        System.out.println(ans);
    }
}